cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java

```diff
@@ -158,24 +158,27 @@ public void removeTask(@NotNull final String taskId, Protos.TaskStatus status) {
         List<CassandraNode> newNodes = new ArrayList<>(nodes.size());
         boolean changed = false;
         for (CassandraNode cassandraNode : nodes) {
-            if (cassandraNode.hasMetadataTask() && cassandraNode.getMetadataTask().getTaskId().equals(taskId)) {
-                // TODO shouldn't we also assume that the server task is no longer running ??
-                // TODO do we need to remove the executor metadata ??
-
-                changed = true;
-                removeExecutorMetadata(cassandraNode.getMetadataTask().getExecutorId());
-                newNodes.add(CassandraNode.newBuilder(cassandraNode)
-                        .clearMetadataTask()
-                        .clearServerTask()
-                        .build());
-            } else if (cassandraNode.hasServerTask() && cassandraNode.getServerTask().getTaskId().equals(taskId)) {
-                changed = true;
-                newNodes.add(CassandraNode.newBuilder(cassandraNode)
-                        .clearServerTask()
-                        .build());
-            } else {
+            CassandraNodeTask nodeTask = CassandraFrameworkProtosUtils.getTaskForNode(cassandraNode, taskId);
+            if (nodeTask == null) {
                 newNodes.add(cassandraNode);
+                continue;
+            }
+            CassandraNode.Builder builder = CassandraFrameworkProtosUtils.removeTask(cassandraNode, nodeTask);
+            changed = true;
+            switch (nodeTask.getTaskType()) {
+                case METADATA:
+                    // TODO shouldn't we also assume that the server task is no longer running ??
```

Comment:
Removing of executor info and taking the server with it should be handled by the status update from `executorLost` so that once `executorLost` is fixed in mesos it can still be handled there.